core/src/test/java/greencity/service/impl/HabitServiceImplTest.java

```diff
@@ -0,0 +1,48 @@
+package greencity.service.impl;
+
+import greencity.ModelUtils;
+import greencity.entity.Habit;
+import greencity.entity.HabitDictionary;
+import greencity.entity.HabitDictionaryTranslation;
+import greencity.repository.HabitDictionaryTranslationRepo;
+import greencity.repository.HabitRepo;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+class HabitServiceImplTest {
+
+    @InjectMocks
+    private HabitServiceImpl habitService;
+
+    @Mock
+    private HabitDictionaryTranslationRepo habitDictionaryTranslationRepo;
+
+    @Mock
+    private HabitRepo habitRepo;
+
+    private Habit habit = ModelUtils.getHabit();
+    private HabitDictionaryTranslation habitDictionaryTranslation = ModelUtils.getHabitDictionaryTranslation();
```

Comment:
Good approach to create testing elements inside each method in case you will need to make different changes when new tests will be added